src/Common/OpenTelemetryTraceContext.h

```diff
@@ -71,7 +72,22 @@ struct Span
     }
 
 private:
-    bool addAttributeImpl(std::string_view name, std::string_view value) noexcept;
+    template <class T>
+    bool addAttributeImpl(std::string_view name, T value) noexcept
+    {
+        try
+        {
+            if constexpr (std::is_same_v<T, std::string_view> || std::is_same_v<T, String> || std::is_same_v<T, const char *>)
+                this->attributes.push_back(Tuple{name, std::move(value)});
+            else
+                this->attributes.push_back(Tuple{name, toString(value)});
```

Comment:
Can you explain how old code led to crash? As we create a new `String` object inside `Field` here (inside `Tuple`), it should be ok to pass temporary `String` object to `addAttributeImpl` even if it has `std::string_view` argument type. Or it's not?